### ISSUE TITLE
Allow users to configure OTel parameters for their excecutor

### ISSUE THREAD

**Issue by calebschoepp** [spinkube/spin-operator#253]
Users need to be able to set OTel parameters like `OTEL_EXPORTER_OTLP_ENDPOINT` on their chosen executor to enable o11y.

**Comment by calebschoepp** [comment]
Relevant design work is being tracked [here](https://github.com/spinkube/skips/pull/4).

**Comment by calebschoepp** [comment]
Talking through this with Michelle we decided it makes sense to define OTEL parameters on the `SpinAppExecutor`, but that it probably makes sense to hold off on defining OTEL parameters on the `SpinApp` (not all executors would be able to support setting it at the app level and we also don't want to pull too many infra concerns into the app CR). We also think that the OTEL values should be scoped under the deployment config since they only work for sure with the shim executor.

We'll allow list the OTEL values in the operator so that arbitrary environment variables can't be injected. Here's an example of the schema.

```
apiVersion: [core.spinoperator.dev/v1alpha1](http://core.spinoperator.dev/v1alpha1)
kind: SpinAppExecutor
metadata:
  name: containerd-shim-spin
spec:
  createDeployment: true
  deploymentConfig:
    runtimeClassName: wasmtime-spin-v2
    installDefaultCACerts: true
    otel: map[string, string]
```

**Comment by calebschoepp** [comment]
#273 is related.

**Comment by calebschoepp** [comment]
@asteurer is working on this

**Comment by calebschoepp** [comment]
Closed by #273
